packages/iris-grid/src/sidebar/visibility-ordering-builder/VisibilityOrderingBuilder.tsx

```diff
@@ -1201,9 +1201,9 @@ class VisibilityOrderingBuilder extends Component<
       treeItems
     );
 
-    const queryParams = {
-      queriedColumnIndex,
-      changeQueriedColumnIndex: (direction: 'forward' | 'back') =>
+    const cursor = {
+      index: queriedColumnIndex as number,
```

Comment:
You shouldn't need this cast, and in fact it's pointing out an error - you could be passing `index: undefined` here, which is not what `SearchInput` is expecting.
Instead, you should just check if it is not set and just return `undefined` for that instead of the parameter, e.g.:
```
    const cursor =
      queriedColumnIndex != null
        ? {
            index: queriedColumnIndex,
            next: (direction: 'forward' | 'back') =>
              this.changeSelectedColumn(direction),
          }
        : undefined;
```

Reply:
I unresolved one of your previous comments. I didn't think it through properly but queriedColumnIndex should be able to be undefined. When the user searches initially there is no index yet. But cursor should still be passed in since we need the next function for the forward and back buttons